types/sinon/ts3.1/index.d.ts

```diff
@@ -29,7 +31,7 @@ declare namespace Sinon {
          * so a call that received the provided arguments (in the same spots) and possibly others as well will return true.
          * @param args
          */
-        calledWith(...args: MatchArguments<TArgs>): boolean;
+        calledWith(...args: Partial<MatchArguments<TArgs>>): boolean;
```

Comment:
Unfortunately this does not work quite as intended. I assume your intention is that you can undersupply `TArgs`, i.e.:

```ts
let spy: SinonSpy<[number, string, boolean], void>;
spy.calledWith(1, "a", true); // fully supplied
spy.calledWith(1, "a"); // undersupplied
spy.calledWith(1); // undersupplied
spy.calledWith(); // undersupplied
```

The problem is that `Partial<[number, string, boolean]>` results in `[number?, string?, boolean?]`. While this *does* allow you to undersupply arguments, keep in mind that 
the types in the tuple are actually expanded to `[(number | undefined)?, (string | undefined)?, (boolean | undefined)?]`, meaning that the following are *also* legal:

```ts
spy.calledWith(undefined, undefined, undefined); // fully supplied but invalid
spy.calledWith(undefined, undefined); // undersupplied but invalid
spy.calledWith(undefined); // undersupplied but invalid
```

Unfortunately, what you probably wanted was `[] | [number] | [number, string] | [number, string, boolean]`. Unfortunately, the type system today does not have a way
to express this. It could be that this is an acceptable risk with regards to the soundness of `calledWith`. It *is* important to note that sinon does distinguish between
`spy.calledWith(1, undefined)` and `spy.calledWith(1)` since it uses `arguments.length` to handle undersupplied arguments, so `spy.calledWith(undefined)` will probably just
return `false`.

Reply:
i was aware of this but it seems until we have a way of representing it in the type system, its the closest solution and is still better than what we had (as what we had is broken, in that valid sinon code will not compile).

if you agree then ill leave it as it is here, its an unfortunate compromise but again is better than what we have currently IMO